src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -239,15 +245,20 @@ private static void EmitFindFirstChar(IndentedTextWriter writer, RegexMethod rm,
             bool rtl = code.RightToLeft;
             bool hasTextInfo = false;
 
+            // In some cases, we need to emit declarations at the beginning of the method, but we only discover we need them later.
+            // To handle that, we emit a placeholder value that's not valid C#, and then at the end of the code generation we either
+            // delete it if no additional declarations are required, or we replace it with the list of additional declarations
+            // built up while generating code.
+            var additionalDeclarations = new HashSet<string>();
+
             // Emit locals initialization
             writer.WriteLine("global::System.ReadOnlySpan<char> runtextSpan = base.runtext;");
             writer.WriteLine("int runtextpos = base.runtextpos;");
-            writer.WriteLine("int runtextend = base.runtextend;");
             if (rtl)
             {
                 writer.WriteLine("int runtextbeg = base.runtextbeg;");
             }
-            writer.WriteLine("int ch;");
```

Comment:
Is there a reason why this was an int before but now (if needed) will be a char? After seeing what it is used for looks like a char is fine, but just wondering if there was a special reason why this had extra bytes before.

Reply:
No reason.  I was quickly translating our RegexCompiler code, which uses an Int32, and didn't fix it until I revisited it here.